packages/drivers/odsp-driver/src/handleReadonlyServerState.ts

```diff
@@ -0,0 +1,56 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ITelemetryLogger } from "@fluidframework/common-definitions";
+import { assert, delay } from "@fluidframework/common-utils";
+import { getRetryDelayFromError } from "@fluidframework/driver-utils";
+import { OdspErrorType } from "@fluidframework/odsp-driver-definitions";
+import { FetchType } from "./epochTracker";
+
+export async function handleReadOnlyServerState<T>(
+    api: () => Promise<T>,
+    logger: ITelemetryLogger,
+    fetchType: FetchType,
+) {
+    let result: T | undefined;
+    let success = false;
+    let retryAfterMs = 1000; // has to be positive!
+    let numRetries = 0;
+    const maxRetries = 5;
```

Comment:
I think we should follow up here (with an issue to consider in the future).
These retries multiplex in non-easily trackable shape. I.e. here we have 5 retries, 409 has its own retries. This makes it hard to control end user experience, as in the worst case we could have 5*3 retries. Usually it's better to have a single place where overall policy RE retries is implemented.